### PR TITLE
Artifact Overview

### PR DESCRIPTION
**Pull request by RDIL**
# New Features

* Yarn
* Artifact Overview (see preview)

## Preview

https://user-images.githubusercontent.com/34555510/103377825-743f4300-4aae-11eb-85e5-fd60f9b8c119.mp4

### CODE REVIEW

**Comment by RDIL** [comment]
Preview:

https://user-images.githubusercontent.com/34555510/103377825-743f4300-4aae-11eb-85e5-fd60f9b8c119.mp4

**Comment by RDIL** [comment]
@fkorotkov review ping 😃

File: src/components/artifacts/ArtifactsView.tsx [hunk 1]
```diff
@@ -192,9 +207,14 @@ class ArtifactsView extends React.Component<Props> {
     return (
       <Paper elevation={1}>
         <Toolbar className={classes.title}>
-          <Typography variant="h6" color="inherit">
+          <Typography variant="h6" color="inherit" className={classes.title}>
             {this._currentPath() || 'Artifacts'}
           </Typography>
+          {this._getSelectedArtifact() === null ? null : (
+            <Button onClick={() => this.setState({ isFolderView: !this.state.isFolderView })}>
```
Comment (fkorotkov) [inline src/components/artifacts/ArtifactsView.tsx:214]: I think here `selectedArtifactName` and `selectedPath` will be overriden. I think `updateState` should be used instead here.

Maybe also use `@material-ui/icons/ViewList` and `@material-ui/icons/AccountView` in a Toggle Button?

Comment (RDIL) [reply src/components/artifacts/ArtifactsView.tsx:214]: Nice find! I'll fix this once I get home.

File: src/components/artifacts/ArtifactsView.tsx [hunk 2]
```diff
@@ -192,9 +208,17 @@ class ArtifactsView extends React.Component<Props> {
     return (
       <Paper elevation={1}>
         <Toolbar className={classes.title}>
-          <Typography variant="h6" color="inherit">
+          <Typography variant="h6" color="inherit" className={classes.title}>
             {this._currentPath() || 'Artifacts'}
           </Typography>
+          {this._getSelectedArtifact() === null ? null : (
+            <Button
```
Comment (fkorotkov) [inline src/components/artifacts/ArtifactsView.tsx:215]: I meant these toggle buttons https://material-ui.com/components/toggle-button/#toggle-buttons

Comment (RDIL) [reply src/components/artifacts/ArtifactsView.tsx:215]: Ohhh, that might help.  I can do this later, I've got a few other projects needing bug fixes in the meantime 😄

Comment (RDIL) [reply src/components/artifacts/ArtifactsView.tsx:215]: Done!

**Comment by RDIL** [comment]
Build fixed.

**Comment by fkorotkov** [comment]
Foxed some missing parts in 7d6f2c16016ceadede15792f2693ccea1f653b5b

**Comment by RDIL** [comment]
Thanks for merging!

**Pull request merged**